c/shared/source/uforead/uforead.c

```diff
@@ -1887,6 +1752,25 @@ static int parseXMLGlifFile(ufoCtx h, xmlNodePtr cur, int tag, unsigned long *un
     return ufoSuccess;
 }
 
+static bool setLibKey(ufoCtx h, char* keyName, xmlNodePtr cur){
+    abfTopDict* top = &h->top;
+    char* keyValue = parseXMLKeyValue(h, cur);
+    if (!keyValueValid(h, cur, keyValue, keyName))
+        return false;
+
+    if (!strcmp(keyName, "public.glyphOrder"))
+       setGlifOrderArray(h, keyName);
+    else if (!strcmp(keyName, "com.adobe.type.cid.CIDFontName"))
+        top->cid.CIDFontName.ptr = copyStr(h, keyValue);
+    else if (!strcmp(keyName, "com.adobe.type.cid.Registry"))
+        top->cid.Registry.ptr = copyStr(h, keyValue);
+    else if (!strcmp(keyName, "com.adobe.type.cid.Ordering"))
+        top->cid.Ordering.ptr = copyStr(h, keyValue);
+    else if (!strcmp(keyName, "com.adobe.type.cid.Supplement"))
+        top->cid.Supplement = atol(copyStr(h, keyValue));
+    return true;
```

Comment:
Should we add an `else assert(false);` before this line, to track that `keyValueValid()` should correspond to this list of strings? (And is that right? If `keyValueValid()` also has keys from other contexts, should we do `else return false;`?)

Reply:
I intended the bool return values mostly for the parser to know if there's any missing items (keys or values) so it wouldn't iterate to the next item assuming it was a new key. Unknown key/value pairs are parseable (simply ignored). 

However, I agree with you that we should return false here for when we add in the warning messages that'll let the users know which keys weren't parsed (with the verbosity flag), so I did add the `else return false;` in for this and `setFontInfoKey`. 

Now, once it returns the false from setXMLFileKey to its caller, it iterates to the cur->next, which is the keyValue (that wasn't set because it was unknown), and then it realizes it's not a key and skips it again to get to the next key/value pair. So returning false for unknown key/value still functions as expected for this addition.

`keyValueValid` only checks if the key value is parsable, so I wouldn't add the assert. I renamed `keyValueValid` to `keyValueParseable` to clarify its purpose.

Thanks for this catch!